File: 3DTouchObjectiveC/3DTouchObjectiveC/PeekViewController.m
//
//  PeekViewController.m
//  3DTouchObjectiveC
//
//  Created by Bao Tran on 11/12/16.
//  Copyright © 2016 Bao Tran. All rights reserved.
//

#import "PeekViewController.h"

@interface PeekViewController ()

@end

@implementation PeekViewController

- (void)viewDidLoad {
    
    [super viewDidLoad];
}

- (void)didReceiveMemoryWarning {
    
    [super didReceiveMemoryWarning];
}

- (void)dismissPeek {
    [self dismissViewControllerAnimated:YES completion:nil];
}

- (void)is3DTouchAvailable {
    if (self.traitCollection.forceTouchCapability != UIForceTouchCapabilityAvailable) {
        
        UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc]initWithTarget:self action:@selector(dismissPeek)];
        
        [self.view addGestureRecognizer:tap];
        
    }
}

@end

File: 3DTouchSwift/3DTouchSwift/PeekViewController.swift
//
//  PeekViewController.swift
//  3DTouchSwift
//
//  Created by Bao Tran on 11/12/16.
//  Copyright © 2016 Bao Tran. All rights reserved.
//

import UIKit

class PeekViewController: UIViewController {

    override func viewDidLoad() {
        
        super.viewDidLoad()
    }

    override func didReceiveMemoryWarning() {
        
        super.didReceiveMemoryWarning()
    }
    
    func dismissVC() {
        
        self.dismissViewControllerAnimated(true, completion: nil)
        
    }
    
    func is3DTouchAvailable() {
        
        if self.traitCollection.forceTouchCapability != UIForceTouchCapability.Available {
            let tap = UITapGestureRecognizer.init(target: self, action: #selector(PeekViewController.dismissVC))
            self.view.addGestureRecognizer(tap)
        }
    }

}
